processor/schemaprocessor/internal/migrate/attributes.go

```diff
@@ -11,24 +11,21 @@ import (
 	"go.uber.org/multierr"
 )
 
-// AttributeChangeSet represents an unscoped entry that can be applied.
-//
+// AttributeChangeSet represents a rename_attributes type operation.
 // The listed changes are duplicated twice
 // to allow for simplified means of transition to or from a revision.
 type AttributeChangeSet struct {
-	updates  ast.AttributeMap
+	// The keys are the old attribute name used in the previous version, the values are the
+	// new attribute name starting from this version (comment from ast.AttributeMap)
+	updates ast.AttributeMap
+	// the inverse of the updates map
 	rollback ast.AttributeMap
```

Comment:
Added.  From the [Schema OTEP](https://github.com/open-telemetry/oteps/blob/main/text/0152-telemetry-schemas.md#all-section) - 
> When the reverse transformation is not possible it is considered an incompatible change. In this case the MAJOR version number of the schema SHOULD be increased in the new version.  

Should the schema processor be trying to enforce this?

Reply:
No, I don't think it is the processor's responsibility to enforce this. Schema authors are responsible for this. 
For troubleshooting purposes the processor may want to log this condition if it is feasible to detect it.

Reply:
Thanks for adding the comments to `updates`. I am still unsure what the `rollback` map contains if the reverse translation ambiguous. Do we avoid creating AttributeChangeSet? Do we detect this situation or it will be a silent incorrect operation if the schema file is incorrect (does not bump the MAJOR version number)?

Reply:
Oops - left a comment lower in the file about that
`	// for ambiguous rollbacks (if updates contains entries with multiple keys that have the same value), rollback contains the last key iterated over in mappings`

We do not detect this currently